### DISCUSSION TITLE
artisan modbus communication error

### DISCUSSION THREAD

**Discussion by yeormcof** [artisan-roaster-scope/artisan#1691] [Q&A] [answered]
Hello, I would like to inquire about modbus communication error occurring in Artisan.
The roasting machine I'm using is a T2. The manufacturer speculates that it's a problem with the laptop or the artisan program. However, the same error occurs when I replace 3 laptops and connect them. The artisan version I'm currently using is 2.8.2. The error message appears, but it's recorded. But when I install the higher version, it doesn't even record. I've tried the latest version of the driver, and I've tried the one provided by the manufacturer, but I get the same error message. I've also replaced the USB cable. Is there a way to fix it?

**Comment by MAKOMO** [comment]
Most likely a configuration issue. Could you please attach your Artisan settings (save via menu Help >> Save Settings into a file named artisan-settings.aset, rename the file into artisan-settings.txt and drag-and-drop it here on GitHub to a comment field) as well as your Artisan log file (click the plus icon in the upper left corner while holding the OPTION key on macOS or ALT key on Windows; an email window will open with the artisan.log file attached  which you please send to the given address). I will take a look.

**Reply by yeormcof** [reply]
[artisan-settings.txt](https://github.com/user-attachments/files/16956904/artisan-settings.txt)
Thanks. One more thing, I'd appreciate it if you could explain again how to send the log file.

**Comment by poundy** [comment]
Two or three other things. 

Please capture the error message you're seeing. The logs as Makomo requests will help immesurably but the picture is a quick way to tell us what you see.

Please explain exactly what you mean "when I install the higher version"? Do you mean a current Artisan version? 

How long has the problem been occurring? Has it ever been reliable and this has just started happening, or are you new to this roaster and Artisan?

**Reply by yeormcof** [reply]
The higher version means the latest version, and it's been about a year since I saw the error message.[
![스크린샷 2024-09-11 105653](https://github.com/user-attachments/assets/8d17e02d-c216-4290-885e-b9e9c9bfc949)

**Reply by poundy** [reply]
you talked about driver versions, please just be clear are you talking about Affinity version or those drivers???

**Reply by yeormcof** [reply]
I was talking about the Artisan program version.

**Reply by yeormcof** [reply]
I've been roasting with Artisan for about 4-5 years.

**Reply by poundy** [reply]
and only now this modbus error is starting? Or is this roaster newly being commissioned?

**Reply by yeormcof** [reply]
It's been 4-5 years since I used this roasting machine with artisan, and it's been about a year since the error message appeared.

**Reply by poundy** [reply]
sorry still don't understand the specific behaviour. Can you elaborate more?  Is the below accurate?

The roaster has been used for the last 4 years with Artisan, and it's worked fine all that time until starting about a year ago, when the error occasionally pops up.

You attempted to upgrade to the latest Artisan but now are not getting any communication (which Marko hinted at being a configuration issue above)

**Reply by yeormcof** [reply]
sorry.
1. I've been using Artisan with the roasting machine I'm currently using for 4-5 years, and I've used it without any problems.
2. A year ago, "modbus communication error" occurred, but the profile was recorded. The artisan version in use is 2.8.2.
3. I tried installing an latest version(artisan) than 2.8.2, but the "connected" and "modbus communication error" messages were repeated and the profile was not recorded.

**Comment by Terracotta-6** [comment]
It's a hardware or configuration problem, Ask the device manufacturer for a communication detection program.

**Reply by yeormcof** [reply]
What does the communication detection program mean?

**Reply by Terracotta-6** [reply]
The problem you describe is not related to the software version, find someone nearby who knows industrial control to help you diagnose the problem on site.

**Reply by yeormcof** [reply]
Thanks for your reply.

**Comment by Terracotta-6** [comment]
Use scan to scan the register address. After scanning, the result is copied to the text file.
![216ef533002e3d90854ae9adb1f51d6](https://github.com/user-attachments/assets/631a0457-a1c1-4192-a1e5-7e651b5606b3)

**Reply by MAKOMO** [reply]
I just received his `artisan.log` file and it indicates that Artisan cannot communicate to his machine at all. Thus there is an issue with the serial setup, USB driver, the USB wire or the machines USB to serial converter. Before this is resolved that MODBUS scan will not work. However, it is a good idea to work out those MODBUS issues he has observed before (most likely a configuration issue) while Artisan was still logging. Anyhow, it is unlikely that his issues are caused by a bug in current Artisan versions.

**Reply by yeormcof** [reply]
thanks for your reply.
How do I fix the serial settings?
And which usb wire and usb to serial converter should I buy?

**Reply by MAKOMO** [reply]
Did you read my PM per email?

**Reply by yeormcof** [reply]
Oh, sorrry. I read it now. My sincere appreciation. And I'll try the things that told me.

**Reply by yeormcof** [reply]
I will send the artisan-settings.aset file within a few days.thanks.

**Comment by MAKOMO** [comment, marked as answer]
The issue was identified. An update of the MODBUS communication library in Artisan v3.0.2 broke the communication with some machines, especially the Trinitas Novustec machines. This has been fixed by now. You can either upgrade to the [continuous build](https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous), or downgrade to [Artisan v3.0.0](https://github.com/artisan-roaster-scope/artisan/releases/tag/v3.0.0) until the next Artisan version is released.

**Comment by yeormcof** [comment]
Thanks a lot :)

보낸 사람: Marko Luther ***@***.***>
날짜: 수요일, 2024년 10월 2일 16:01
받는 사람: artisan-roaster-scope/artisan ***@***.***>
참조: yeormcof ***@***.***>, Author ***@***.***>
주제: Re: [artisan-roaster-scope/artisan] artisan modbus communication error (Discussion #1691)

Closed #1691<https://github.com/artisan-roaster-scope/artisan/discussions/1691> as resolved.

**Reply by MAKOMO** [reply]
Thanks for your support on ironing this out!

**Comment by Terracotta-6** [comment]
lv_baozzhu


***@***.***







电子名片新出VIP模板啦，快来体验>>




扫一扫,


快速添加名片到手机




--------------------------------------------------------------------------------





------------------ 原始邮件 ------------------




 
 发件人: Marko Luther  ***@***.***>
 

 
 发送时间: 2024-10-02 15:01:41
 

 
 收件人:artisan-roaster-scope/artisan ***@***.***> 

 
 抄送:lv_baozzhu ***@***.***>,Comment ***@***.***> 

 
 主题: Re: [artisan-roaster-scope/artisan] artisan modbus communicationerror (Discussion #1691)
 





 The issue was identified. An update of the MODBUS communication library in Artisan v3.0.2 broke the communication with some machines, especially the Trinitas Novustec machines. This has been fixed by now. You can either upgrade to the continuous build, or downgrade to Artisan v3.0.0 until the next Artisan version is released.

 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: ***@***.***>

 






100 15:31:36.365 Connected via MODBUS

99 15:31:35.498 Modbus Communication Error

98 15:31:35.472 MODBUS disconnected

97 15:31:10.202 Connected via MODBUS

96 15:31:09.431 Modbus Communication Error

95 15:31:09.410 MODBUS disconnected

94 15:30:50.203 Connected via MODBUS

93 15:30:49.389 Modbus Communication Error

92 15:30:49.367 MODBUS disconnected

91 15:30:16.206 Connected via MODBUS

90 15:30:15.301 Modbus Communication Error

89 15:30:15.280 MODBUS disconnected

88 15:30:04.203 Connected via MODBUS

87 15:30:03.442 Modbus Communication Error

86 15:30:03.420 MODBUS disconnected

85 15:29:50.203 Connected via MODBUS

84 15:29:49.303 Modbus Communication Error

83 15:29:49.281 MODBUS disconnected

82 15:29:26.203 Connected via MODBUS

81 15:29:25.400 Modbus Communication Error

80 15:29:25.377 MODBUS disconnected

79 15:29:24.203 Connected via MODBUS

78 15:29:23.396 Modbus Communication Error

77 15:29:23.374 MODBUS disconnected

76 15:29:22.204 Connected via MODBUS

75 15:29:21.369 Modbus Communication Error

74 15:29:21.347 MODBUS disconnected

73 15:29:18.203 Connected via MODBUS

72 15:29:17.390 Modbus Communication Error

71 15:29:17.369 MODBUS disconnected

70 15:28:38.204 Connected via MODBUS

69 15:28:37.401 Modbus Communication Error

68 15:28:37.381 MODBUS disconnected

67 15:28:22.203 Connected via MODBUS

66 15:28:21.330 Modbus Communication Error

65 15:28:21.297 MODBUS disconnected

64 15:28:04.203 Connected via MODBUS

63 15:28:03.358 Modbus Communication Error

62 15:28:03.337 MODBUS disconnected

61 15:27:54.204 Connected via MODBUS

60 15:27:53.283 Modbus Communication Error

59 15:27:53.253 MODBUS disconnected

58 15:27:50.204 Connected via MODBUS

57 15:27:49.379 Modbus Communication Error

56 15:27:49.340 MODBUS disconnected

55 15:27:48.972 Serial Exception: Unable to open serial port

54 15:27:48.944 Scope monitoring...

53 15:27:48.485 Scope has been reset

52 15:27:41.895 Connected via MODBUS

51 15:27:41.751 Connected via MODBUS

50 15:27:41.729 PID mode manual

49 15:27:41.681 MODBUS disconnected

48 15:27:40.635 Scope stopped

47 15:27:39.368 Alarm 12 triggered

46 15:27:38.973 Scope recording stopped

45 15:27:38.942 Modbus Communication Error

44 15:27:38.835 MODBUS disconnected

43 15:27:38.784 PID mode manual

42 15:27:38.744 Roast ended at 00:58 BT = 25.6 C

41 15:27:31.691 Connected via MODBUS

40 15:27:30.916 Modbus Communication Error

39 15:27:30.834 MODBUS disconnected

38 15:27:29.687 Connected via MODBUS

37 15:27:28.849 MODBUS disconnected

36 15:27:28.827 Modbus Communication Error

35 15:27:23.685 Connected via MODBUS

34 15:27:22.867 Modbus Communication Error

33 15:27:22.775 MODBUS disconnected

32 15:27:21.686 Connected via MODBUS

31 15:27:20.915 Modbus Communication Error

30 15:27:20.825 MODBUS disconnected

29 15:27:09.714 Connected via MODBUS

28 15:27:09.042 MODBUS disconnected

27 15:27:08.981 Modbus Communication Error

26 15:27:07.688 Connected via MODBUS

25 15:27:07.003 Modbus Communication Error

24 15:27:06.865 MODBUS disconnected

23 15:27:05.692 Connected via MODBUS

22 15:27:04.916 Modbus Communication Error

21 15:27:04.761 MODBUS disconnected

20 15:27:01.691 Connected via MODBUS

19 15:27:00.854 Modbus Communication Error

18 15:27:00.740 MODBUS disconnected

17 15:26:59.687 Connected via MODBUS

16 15:26:58.998 Modbus Communication Error

15 15:26:58.868 MODBUS disconnected

14 15:26:51.710 Connected via MODBUS

13 15:26:50.852 Modbus Communication Error

12 15:26:50.742 MODBUS disconnected

11 15:26:45.706 Connected via MODBUS

10 15:26:44.876 Modbus Communication Error

9 15:26:44.774 MODBUS disconnected

8 15:26:43.688 Connected via MODBUS

7 15:26:42.855 Modbus Communication Error

6 15:26:42.743 MODBUS disconnected

5 15:26:40.466 Roast time starts now 00:00 BT = 25.8C

4 15:26:31.700 Connected via MODBUS

3 15:26:30.921 Modbus Communication Error

2 15:26:30.837 MODBUS disconnected

1 15:26:29.689 Connected via MODBUS


version = 3.0.3

108 15:35:18.618 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2041 || Code = 4 || Rx = 65486 || retries = 0

107 15:35:18.618 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2230 || Code = 3 || Rx = 0 || retries = 0

106 15:35:18.618 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2041 || Code = 4 || Rx = 1050 || retries = 0

105 15:35:18.618 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2230 || Code = 3 || Rx = 1000 || retries = 0

104 15:35:18.618 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2000 || Code = 4 || Rx = 252 || retries = 0

103 15:35:18.618 MODBUS readSingleRegister : 1.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2000 || Code = 4 || Rx = 260 || retries = 0

102 15:35:18.617 cache reading : Slave = 2 || Register = 2230 || Rx = 0

101 15:35:18.617 MODBUS readActiveregisters : 82.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2230 || Code = 3 || Rx# = 1

100 15:35:18.534 cache reading : Slave = 1 || Register = 2230 || Rx = 1000

99 15:35:18.534 MODBUS readActiveregisters : 81.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2230 || Code = 3 || Rx# = 1

98 15:35:18.453 cache reading : Slave = 2 || Register = 2041 || Rx = 65486

97 15:35:18.453 MODBUS readActiveregisters : 80.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2041 || Code = 4 || Rx# = 1

96 15:35:18.373 cache reading : Slave = 2 || Register = 2000 || Rx = 252

95 15:35:18.373 MODBUS readActiveregisters : 75.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2000 || Code = 4 || Rx# = 1

94 15:35:18.298 cache reading : Slave = 1 || Register = 2041 || Rx = 1050

93 15:35:18.298 MODBUS readActiveregisters : 73.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2041 || Code = 4 || Rx# = 1

92 15:35:18.225 cache reading : Slave = 1 || Register = 2000 || Rx = 260

91 15:35:18.225 MODBUS readActiveregisters : 39.2ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2000 || Code = 4 || Rx# = 1

90 15:35:16.590 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2041 || Code = 4 || Rx = 65486 || retries = 0

89 15:35:16.589 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2230 || Code = 3 || Rx = 0 || retries = 0

88 15:35:16.589 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2041 || Code = 4 || Rx = 1050 || retries = 0

87 15:35:16.589 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2230 || Code = 3 || Rx = 1000 || retries = 0

86 15:35:16.589 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2000 || Code = 4 || Rx = 252 || retries = 0

85 15:35:16.589 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2000 || Code = 4 || Rx = 260 || retries = 0

84 15:35:16.588 cache reading : Slave = 2 || Register = 2230 || Rx = 0

83 15:35:16.588 MODBUS readActiveregisters : 67.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2230 || Code = 3 || Rx# = 1

82 15:35:16.521 cache reading : Slave = 1 || Register = 2230 || Rx = 1000

81 15:35:16.521 MODBUS readActiveregisters : 66.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2230 || Code = 3 || Rx# = 1

80 15:35:16.455 cache reading : Slave = 2 || Register = 2041 || Rx = 65486

79 15:35:16.455 MODBUS readActiveregisters : 78.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2041 || Code = 4 || Rx# = 1

78 15:35:16.377 cache reading : Slave = 2 || Register = 2000 || Rx = 252

77 15:35:16.377 MODBUS readActiveregisters : 80.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2000 || Code = 4 || Rx# = 1

76 15:35:16.297 cache reading : Slave = 1 || Register = 2041 || Rx = 1050

75 15:35:16.297 MODBUS readActiveregisters : 63.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2041 || Code = 4 || Rx# = 1

74 15:35:16.234 cache reading : Slave = 1 || Register = 2000 || Rx = 260

73 15:35:16.234 MODBUS readActiveregisters : 47.7ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2000 || Code = 4 || Rx# = 1

72 15:35:14.523 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2041 || Code = 4 || Rx = 65486 || retries = 0

71 15:35:14.523 MODBUS readSingleRegister : 1.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2230 || Code = 3 || Rx = 0 || retries = 0

70 15:35:14.522 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2041 || Code = 4 || Rx = 1050 || retries = 0

69 15:35:14.522 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2230 || Code = 3 || Rx = 1000 || retries = 0

68 15:35:14.522 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2000 || Code = 4 || Rx = 252 || retries = 0

67 15:35:14.522 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2000 || Code = 4 || Rx = 260 || retries = 0

66 15:35:14.522 cache reading : Slave = 2 || Register = 2230 || Rx = 0

65 15:35:14.522 MODBUS readActiveregisters : 54.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2230 || Code = 3 || Rx# = 1

64 15:35:14.468 cache reading : Slave = 1 || Register = 2230 || Rx = 1000

63 15:35:14.468 MODBUS readActiveregisters : 63.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2230 || Code = 3 || Rx# = 1

62 15:35:14.405 cache reading : Slave = 2 || Register = 2041 || Rx = 65486

61 15:35:14.405 MODBUS readActiveregisters : 49.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2041 || Code = 4 || Rx# = 1

60 15:35:14.356 cache reading : Slave = 2 || Register = 2000 || Rx = 252

59 15:35:14.356 MODBUS readActiveregisters : 63.2ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2000 || Code = 4 || Rx# = 1

58 15:35:14.293 cache reading : Slave = 1 || Register = 2041 || Rx = 1050

57 15:35:14.293 MODBUS readActiveregisters : 48.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2041 || Code = 4 || Rx# = 1

56 15:35:14.245 cache reading : Slave = 1 || Register = 2000 || Rx = 260

55 15:35:14.245 MODBUS readActiveregisters : 58.7ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2000 || Code = 4 || Rx# = 1

54 15:35:12.470 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2041 || Code = 4 || Rx = 65486 || retries = 0

53 15:35:12.470 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2230 || Code = 3 || Rx = 0 || retries = 0

52 15:35:12.470 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2041 || Code = 4 || Rx = 1050 || retries = 0

51 15:35:12.470 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2230 || Code = 3 || Rx = 1000 || retries = 0

50 15:35:12.470 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2000 || Code = 4 || Rx = 252 || retries = 0

49 15:35:12.470 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2000 || Code = 4 || Rx = 260 || retries = 0

48 15:35:12.469 cache reading : Slave = 2 || Register = 2230 || Rx = 0

47 15:35:12.469 MODBUS readActiveregisters : 36.9ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2230 || Code = 3 || Rx# = 1

46 15:35:12.432 cache reading : Slave = 1 || Register = 2230 || Rx = 1000

45 15:35:12.432 MODBUS readActiveregisters : 46.6ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2230 || Code = 3 || Rx# = 1

44 15:35:12.386 cache reading : Slave = 2 || Register = 2041 || Rx = 65486

43 15:35:12.385 MODBUS readActiveregisters : 48.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2041 || Code = 4 || Rx# = 1

42 15:35:12.337 cache reading : Slave = 2 || Register = 2000 || Rx = 252

41 15:35:12.337 MODBUS readActiveregisters : 46.9ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2000 || Code = 4 || Rx# = 1

40 15:35:12.289 cache reading : Slave = 1 || Register = 2041 || Rx = 1050

39 15:35:12.289 MODBUS readActiveregisters : 47.5ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2041 || Code = 4 || Rx# = 1

38 15:35:12.241 cache reading : Slave = 1 || Register = 2000 || Rx = 260

37 15:35:12.241 MODBUS readActiveregisters : 55.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2000 || Code = 4 || Rx# = 1

36 15:35:10.518 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2041 || Code = 4 || Rx = 65486 || retries = 0

35 15:35:10.504 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2230 || Code = 3 || Rx = 0 || retries = 0

34 15:35:10.504 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2041 || Code = 4 || Rx = 1050 || retries = 0

33 15:35:10.490 MODBUS readSingleRegister : 1.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2230 || Code = 3 || Rx = 1000 || retries = 0

32 15:35:10.477 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2000 || Code = 4 || Rx = 252 || retries = 0

31 15:35:10.464 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2000 || Code = 4 || Rx = 260 || retries = 0

30 15:35:10.464 cache reading : Slave = 2 || Register = 2230 || Rx = 0

29 15:35:10.464 MODBUS readActiveregisters : 47.8ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2230 || Code = 3 || Rx# = 1

28 15:35:10.416 cache reading : Slave = 1 || Register = 2230 || Rx = 1000

27 15:35:10.416 MODBUS readActiveregisters : 45.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2230 || Code = 3 || Rx# = 1

26 15:35:10.370 cache reading : Slave = 2 || Register = 2041 || Rx = 65486

25 15:35:10.370 MODBUS readActiveregisters : 47.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2041 || Code = 4 || Rx# = 1

24 15:35:10.323 cache reading : Slave = 2 || Register = 2000 || Rx = 252

23 15:35:10.323 MODBUS readActiveregisters : 46.3ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2000 || Code = 4 || Rx# = 1

22 15:35:10.277 cache reading : Slave = 1 || Register = 2041 || Rx = 1050

21 15:35:10.277 MODBUS readActiveregisters : 47.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2041 || Code = 4 || Rx# = 1

20 15:35:10.230 cache reading : Slave = 1 || Register = 2000 || Rx = 260

19 15:35:10.230 MODBUS readActiveregisters : 43.7ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2000 || Code = 4 || Rx# = 1

18 15:35:08.477 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2041 || Code = 4 || Rx = 65486 || retries = 0

17 15:35:08.470 MODBUS readSingleRegister : 1.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2230 || Code = 3 || Rx = 0 || retries = 0

16 15:35:08.469 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2041 || Code = 4 || Rx = 1050 || retries = 0

15 15:35:08.462 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2230 || Code = 3 || Rx = 1000 || retries = 0

14 15:35:08.462 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2000 || Code = 4 || Rx = 252 || retries = 0

13 15:35:08.456 MODBUS readSingleRegister : 0.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2000 || Code = 4 || Rx = 260 || retries = 0

12 15:35:08.451 cache reading : Slave = 2 || Register = 2230 || Rx = 0

11 15:35:08.451 MODBUS readActiveregisters : 42.4ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2230 || Code = 3 || Rx# = 1

10 15:35:08.403 cache reading : Slave = 1 || Register = 2230 || Rx = 1000

9 15:35:08.403 MODBUS readActiveregisters : 31.0ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2230 || Code = 3 || Rx# = 1

8 15:35:08.372 cache reading : Slave = 2 || Register = 2041 || Rx = 65486

7 15:35:08.372 MODBUS readActiveregisters : 59.7ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2041 || Code = 4 || Rx# = 1

6 15:35:08.309 cache reading : Slave = 2 || Register = 2000 || Rx = 252

5 15:35:08.309 MODBUS readActiveregisters : 45.8ms => COM5,115200,8,N,1,1.0 || Slave = 2 || Register = 2000 || Code = 4 || Rx# = 1

4 15:35:08.263 cache reading : Slave = 1 || Register = 2041 || Rx = 1050

3 15:35:08.263 MODBUS readActiveregisters : 30.3ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2041 || Code = 4 || Rx# = 1

2 15:35:08.232 cache reading : Slave = 1 || Register = 2000 || Rx = 260

1 15:35:08.231 MODBUS readActiveregisters : 44.9ms => COM5,115200,8,N,1,1.0 || Slave = 1 || Register = 2000 || Code = 4 || Rx# = 1

**Reply by MAKOMO** [reply]
??
